src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb

```diff
@@ -0,0 +1,450 @@
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Microsoft.CodeAnalysis.VisualBasic.UnitTests.Emit
+Imports Roslyn.Test.Utilities
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.UnitTests
+    Public Class CodeGenTuples
+        Inherits BasicTestBase
+
+        <Fact()>
+        Public Sub TupleTypeBinding()
+
+            Dim verifier = CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+Imports System
+Module C
+
+    Sub Main()
+        Dim t as (Integer, Integer)
+        console.writeline(t)            
+    End Sub
+End Module
+
+Namespace System
+    Structure ValueTuple(Of T1, T2)
+        Public Overrides Function ToString() As String
+            Return "hello"
+        End Function
+    End Structure
+End Namespace
+
+    </file>
+</compilation>, expectedOutput:=<![CDATA[
+hello
+            ]]>)
+
+            verifier.VerifyIL("C.Main", <![CDATA[
+{
+  // Code size       12 (0xc)
+  .maxstack  1
+  .locals init (System.ValueTuple(Of Integer, Integer) V_0) //t
+  IL_0000:  ldloc.0
+  IL_0001:  box        "System.ValueTuple(Of Integer, Integer)"
+  IL_0006:  call       "Sub System.Console.WriteLine(Object)"
+  IL_000b:  ret
+}
+]]>)
+        End Sub
+
+        <Fact()>
+        Public Sub TupleFieldBinding()
+
+            Dim verifier = CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+Imports System
+Module C
+
+    Sub Main()
+        Dim t as (Integer, Integer)
+
+        t.Item1 = 42
+        t.Item2 = t.Item1
+        console.writeline(t.Item2)            
+    End Sub
+End Module
+
+Namespace System
+    Structure ValueTuple(Of T1, T2)
+        Public Item1 As T1
+        Public Item2 As T2
+    End Structure
+End Namespace
+
+    </file>
+</compilation>, expectedOutput:=<![CDATA[
+42
+            ]]>)
+
+            verifier.VerifyIL("C.Main", <![CDATA[
+{
+  // Code size       34 (0x22)
+  .maxstack  2
+  .locals init (System.ValueTuple(Of Integer, Integer) V_0) //t
+  IL_0000:  ldloca.s   V_0
+  IL_0002:  ldc.i4.s   42
+  IL_0004:  stfld      "System.ValueTuple(Of Integer, Integer).Item1 As Integer"
+  IL_0009:  ldloca.s   V_0
+  IL_000b:  ldloc.0
+  IL_000c:  ldfld      "System.ValueTuple(Of Integer, Integer).Item1 As Integer"
+  IL_0011:  stfld      "System.ValueTuple(Of Integer, Integer).Item2 As Integer"
+  IL_0016:  ldloc.0
+  IL_0017:  ldfld      "System.ValueTuple(Of Integer, Integer).Item2 As Integer"
+  IL_001c:  call       "Sub System.Console.WriteLine(Integer)"
+  IL_0021:  ret
+}
+]]>)
+        End Sub
+
+        <Fact()>
+        Public Sub TupleFieldBindingLong()
+
+            Dim verifier = CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+Imports System
+Module C
+
+    Sub Main()
+        Dim t as (Integer, Integer, Integer, integer, integer, integer, integer, integer, integer, Integer, Integer, Integer, integer, integer, integer, integer, integer, integer)
```

Comment:
good idea
Fixed

---
In reply to: [71417163](https://github.com/dotnet/roslyn/pull/12596#discussion_r71417163) [](ancestors = 71417163)